infra/rooch-portal-v2/contract/gas_market/sources/gas_airdrop.move

```diff
@@ -0,0 +1,139 @@
+module gas_market::gas_airdrop {
+
+    use std::vector;
+
+    use moveos_std::signer;
+    use moveos_std::object::{Self, Object, ObjectID, to_shared};
+    use moveos_std::tx_context::{sender};
+    use moveos_std::table::{Self, Table};
+
+    use rooch_framework::coin_store::CoinStore;
+    use rooch_framework::gas_coin::RGas;
+    use rooch_framework::coin_store;
+    use rooch_framework::account_coin_store;
+
+    use bitcoin_move::utxo::{Self, UTXO};
+
+    use gas_market::gas_market::AdminCap;
+
+    const INIT_GAS_AMOUNT: u256 = 5000000_00000000;
+    const ONE_RGAS: u256 = 1_00000000;
+
+    //0.01 BTC
+    const SAT_LEVEL_ONE: u64 = 1000000;
+    //0.1 BTC
+    const SAT_LEVEL_TWO: u64 = 10000000;
+
+    const ErrorAirdropNotOpen: u64 = 0;
+    const ErrorInvalidUTXO: u64 = 1;
+    const ErrorAirdropNotEnoughRGas: u64 = 2;
+    const ErrorAlreadyClaimed: u64 = 3;
+    const ErrorUTXOValueIsZero: u64 = 4;
+
+
+    struct RGasAirdrop has key{
+        rgas_store: Object<CoinStore<RGas>>,
+        claim_records: Table<address, u256>, 
+        is_open: bool
+    }
+
+
+    fun init(sender: &signer) {
+      let sender_addr = signer::address_of(sender);
+      let rgas_store = coin_store::create_coin_store<RGas>();
+      let rgas_balance = account_coin_store::balance<RGas>(sender_addr);
+      let airdrop_gas_amount = if(rgas_balance > INIT_GAS_AMOUNT) {
+        INIT_GAS_AMOUNT
+      } else {
+        rgas_balance/3
+      };
+      Self::deposit_to_rgas_store(sender, &mut rgas_store, airdrop_gas_amount);
+      let rgas_airdrop_obj = object::new_named_object(RGasAirdrop{
+          rgas_store,
+          claim_records: table::new(),
+          is_open: true
+      });
+      to_shared(rgas_airdrop_obj)
+    }
+
+    /// Anyone can call this function to help the claimer claim the airdrop
+    public entry fun claim(airdrop_obj: &mut Object<RGasAirdrop>, claimer: address, utxo_ids: vector<ObjectID>){      
```

Comment:
Where is the height snapshot determined?

Reply:
We do not check the snapshot; any address that has UTXO can claim the gas coin.